ui/packages/catalog-ui-search/src/main/webapp/js/model/List.js

```diff
@@ -86,6 +112,47 @@ module.exports = Backbone.AssociatedModel.extend(
         this.updateQuery
       )
     },
+    set(data, ...args) {
+      if (typeof data === 'object') {
+        // for backwards compatability
+        if (data['list.cql'] && !data['list.filters']) {
+          try {
+            const filterTree = CQLUtils.transformCQLToFilter(data['list.cql'])
+            data['list.filters'] = simplifyListFilter(filterTree)
+          } catch (e) {
+            console.log('Invalid cql: ' + data['list.cql'])
+          }
+          data['list.cql'] = undefined
+        } else if (
+          data['list.filters'] &&
+          typeof data['list.filters'] === 'string'
+        ) {
+          try {
+            data['list.filters'] = JSON.parse(data['list.filters'])
+          } catch (e) {
+            data['list.filters'] = undefined
+          }
+        }
+      } else if (data === 'list.filters' && args.length == 1) {
+        args = [simplifyListFilter(args[0])]
+      }
+      if (!IGNORED_DEPRECATED_ATTRIBUTES.includes(data)) {
+        return Backbone.AssociatedModel.prototype.set.call(this, data, ...args)
+      }
+    },
+    get(attr) {
+      if (!IGNORED_DEPRECATED_ATTRIBUTES.includes(attr)) {
+        return Backbone.AssociatedModel.prototype.get.call(this, attr)
+      }
+    },
```

Comment:
❗ If we handle the `list.cql` backwards compatibility case in the `initialize` method, I think we will be able to simplify a lot of this logic or remove it completely. Do you think there will be any issues doing it this way?
```
initialize() {
    if model has list.cql and not list.filters
        convertCqlToFilters()        
    }

    // rest of initialize
}
```

Or we might want to think about updating the backend to do the conversion for us so that the ui only ever has to worry about `list.filters`